trl/trainer/sft_trainer.py

```diff
@@ -390,6 +401,10 @@ def _prepare_dataset(
                 append_concat_token,
                 add_special_tokens,
             )
+        elif non_contaminated_packing:
```

Comment:
I think we should move **checking** `non_contaminated_packing` this to line 380 or we will never reach this line.
Should be like this:

```
if non_contaminated_packing:
      return self._prepare_non_contaminated_packed_dataloader(
          tokenizer, dataset, dataset_text_field, max_seq_length, add_special_tokens
      )

if not packing:
      return self._prepare_non_packed_dataloader(
          tokenizer,
          dataset,
          dataset_text_field,
          max_seq_length,
          formatting_func,
          add_special_tokens,
          remove_unused_columns,
      )

else:
      return self._prepare_packed_dataloader(
          tokenizer,
          dataset,
          dataset_text_field,
          max_seq_length,
          num_of_sequences,
          chars_per_token,
          formatting_func,
          append_concat_token,
          add_special_tokens,
      )
```

Reply:
Ah yes of course, thanks!